Usagov 429 states left nav

IsabelLaurenceau opened the pull request:
<!--- Provide a general summary of your changes in the title above -->
## Jira Task
<!--- Provide a link to the Jira ticket -->
https://cm-jira.usa.gov/browse/USAGOV-429
&&
https://cm-jira.usa.gov/browse/USAGOV-396

## Description
<!--- Describe your changes in detail -->
Files changed: 
web/themes/custom/usagov/templates/breadcrumb.html.twig=
web/themes/custom/usagov/templates/menu--agency-sidebar.html.twig
web/themes/custom/usagov/templates/menu--es-agency-sidebar.html.twig
web/themes/custom/usagov/templates/menu--es-federal-agencies-sidebar.html.twig
web/themes/custom/usagov/templates/menu--federal-agencies-sidebar.html.twig
web/themes/custom/usagov/templates/menu--sidebar_first.html.twig


**Need to rebuild theme registry and flush all caches
“Testing should include removing the menu settings from the agency-index and Agency Placeholder pages -- the goal is to get the pages to look right without putting them all in the menu system.”

Testing on /agency-index & /states: Left menu should resemble formatting of other about-us pages Left menu should have correct links
Breadcrumbs should go up page (with last page not clickable) 

Testing on /agencies/agency-placeholder: Left menu should resemble formatting of other about-us pages Left menu should have correct links
Breadcrumbs should go up page (with last page not clickable) 

**Pages need to be created to test for /state/{name} , /es/indice-agencias , /gobiernos-estatales , /es/agencias & /es/estados


## Checklist for the Developer
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask for help! -->
- [x] A link to the JIRA ticket has been added above.
- [x] The JIRA ticket identifies the desired result of this change.
- [x] The JIRA ticket contains clear acceptance criteria.
- [x] The JIRA ticket contains clear testing steps.
- [x] The JIRA ticket contains a description of "Done".
- [x] Any preparation/installation/update steps required for this code to work properly (drush commands, scripts, configuration updates) are documented in the ticket.

## Checklist for the Peer Reviewers
- [x] The branch name of this PR matches the project standards.
- [x] QA steps are followed and any changes to process are updated in the ticket.
- [x] Code Standards are followed, there are no bad practices in use.
- [x] Config changes (if any) include only necessary changes.
- [x] No errors in Drupal or Client Browser.
- [x] Code has been tested locally (if possible).
- [ ] Following AC and Testing Steps verify changes work as expected.
- [x] There are no known side-effects outside of expected behavior.

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: HardCoded Static Left Nav Mainly understanding what the different classes are doing

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Menu completely hard coded but functional Need to move it and set condition on when to call it

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Moved into separate file Need to 1) Figure out how to call 2) Make not hard-coded

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Correctly calling static page only on agency path

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Getting the menu titles need to get the href -- Also need to get grandparents for specific agency

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Left Menu correct for /agency-index Need to do it for /agencies/agency-placeholder

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Works by calling two new files

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Renamed files

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Static

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Update settings.php for PR

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Merge branch 'dev' into USAGOV-396-federal-agencies-left-nav-menu # Please enter a commit message to explain why this merge is necessary, # especially if it merges an updated upstream into a topic branch. # # Lines starting with '#' will be ignored, and an empty message aborts # the commit.

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Getting correct data need to check formatting

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: removed unnecessary repeat

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Added meta information

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Added breadcrumbs and sidebar menu for es

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Merge branch 'dev' into USAGOV-396-federal-agencies-left-nav-menu

IsabelLaurenceau committed:
USAGOV-396-federal-agencies-left-nav-menu: Added states to the left nav menu

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: Left nav for states should be correct Not sure how to test

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: States breadcrumbs are correct

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: Spanish breadcrumb

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: Spanish breadcrumbs

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: Fixed variable check for federal dir left menu

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: Bring back menu for es

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: removed extra p tag

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: Merge branch 'dev' into USAGOV-429-states-left-nav # Conflicts: # web/themes/custom/usagov/templates/menu--sidebar_first.html.twig

akf commented on web/themes/custom/usagov/templates/breadcrumb.html.twig:
```diff
@@ -7,36 +7,208 @@
-                </li>
-            {% endfor %}
+{% if path('<current>') == '/agency-index' or path('<current>') matches '#\/agencies\/*#'%}
+    <nav class="margin-top-neg-3 mobile-lg:margin-top-2 tablet:margin-top-neg-3 desktop:margin-top-0 margin-bottom-2 mobile-lg:margin-bottom-6" role="navigation" aria-labelledby="system-breadcrumb" >
```
Consider making use of include files for these chunks of static (I think) nav content. I think that would make it easier to understand this file at a glance.

akf replied:
Just like you did with the sidebar! (Also, maybe this just looks like a long chunk of text because I'm looking at it in a diff window.)

IsabelLaurenceau replied:
This was actually my intention however I was not able to get the include to work for the breadcrumbs. I wasn't sure if that was because breadcrumbs work a different way but I can try again!

IsabelLaurenceau replied:
Found it! Had to use include directory ~ not just include

akf commented on web/themes/custom/usagov/templates/menu--sidebar_first.html.twig:
```diff
@@ -15,7 +15,25 @@ A short summary of the behavior of the sidebar nav:
 {% import _self as menus %}
 
-{{ menus.explore(items, 0, null, null) }}
+{# {% set relatedPages = drupal_entity('node', items.url.routeParameters.node )['#node'].field_related_pages %} #}
```
Looks like this was a start at something we don't need?

IsabelLaurenceau replied:
That comment was there when I pulled. I think it was a leftover from when chris was working on the false children.

akf reviewed (changes requested):
The comments I made about specific bits of code are incidental. I saw some issues in testing, several of which would be addressed by a slight change of approach: We should put the "index" pages (/agency-index, /states, and their Spanish equivalents) back in the menu system and not hard-code the menus on those pages. I had been thinking we'd want to hard-code those, and I probably communicated that, but I was wrong! If we put those pages in the menu system, they will appear as desired *so long as* we don't put any of their children in the menu system. 

In addition to removing a couple of special cases from the code, this will fix:
- "index" pages not appearing in the left nav menu for their siblings
- Federal agency index card disappeared from the "About the US ..." page. 

Another thing: the /gobiernos path must have been specified by mistake; it would have to be /es/gobiernos. I have just learned that if I set the language of a page to Spanish, Drupal automatically prepends /es to its path. 

The rest of what I found in testing is below. 

These pages look good, but see above: 

- /agency-index
- /states (BUT, the title should be "State Governments" rather than "State Government")

These look good: 

- /agencies/agency-placeholder
- /es/estados/oregon
- /gobiernos (but it should be /es/gobiernos)

Problems: 

- Left nav menu is missing from /states/oregon (breadcrumbs are good, though) 
- Left nav menu is missing from /es/indice-agencias (but again, breadcrumbs are good) 
- /es/agencias/espanol-example looks almost right to me, but the page title is appearing in the left nav under "Agencias estatales" rather than "Agencias federales". Again, the breadcrumbs have it right. 
- Siblings of /agency-index (like /branches-of-government) don't show the Federal and State directory links in the left nav.
- We lost the "card" for the /agency-index page on /about-the-us

Those last two problems will be solved by putting the "index" pages back into the menu system. I didn't build out enough of the Spanish pages to look at the issues there, but I expect them to be similar. 

The pages that I found to be missing the left nav menu make me wonder whether I set them up wrong somehow, as the pattern isn't obvious.

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: Merge branch 'dev' into USAGOV-429-states-left-nav

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: Removed hard coded index pages

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: Plural for state governments and removed unused files

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: Fixed /es/agencias/espanol-example

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: Fixed - left nav menu is missing from /states/oregon

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: breadcrumbs now call separate file

IsabelLaurenceau commented:
What’s changed:
* sidebar now says "State Governments" rather than "State Government"
* Left nav menu is now showing for /states/oregon 
* /es/agencias/espanol-example is appearing in the left nav under "Agencias federales" rather than “Agencias estatales" 
* Breadcrumbs now use include directory
* "index" pages (/agency-index, /states, and their Spanish equivalents) are back in the menu system and not hard-code the menus on those pages. This should fix: 
    * Siblings showing agencies in side bar
    * Card in the about-us page
    * Left nav missing from /es/indice-agencias
    * /gobiernos (but it should be /es/gobiernos) since it’s no longer hard coded

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: Update breadcrumb.html.twig

akf committed:
Merge branch 'dev' into USAGOV-429-states-left-nav

akf commented on web/themes/custom/usagov/templates/menu--sidebar_first.html.twig:
```diff
@@ -15,7 +15,22 @@ A short summary of the behavior of the sidebar nav:
+{% elseif path('<current>') matches '#\/states\/*#' %}    
+  {% include 'menu--agency-sidebar.html.twig' %}
+
+{% elseif path('<current>') matches '#\/es\/agencias\/*#' or path('<current>') matches '#\/es\/estados\/*#'%}    
```
While you're making updates, why not combine lines 21 and 24 the way you did on line 27 here?

akf replied:
Oops! I just realized that we actually want to take out the clauses for '#\/agencies\/*#' and '#\/states\/*#' -- /agencies isn't actually a page, and /states should have the dynamically-generated menu (which would be why I'm seeing "State Governments" twice on /states now).

akf replied:
Ah, I see -- I was misreading the regexps the first time around.

akf reviewed (changes requested):
This is looking much better. I caught a couple of additional things: 

* The Spanish nav menu links "Agencias estatales" to /es/indice-agencias-estatales, but it should be /es/gobiernos-estatales
* On /states, the left nav shows "State Governments" (bold) and another "State Governments" as a child of the first, in red. I think I found what's causing that and commented on the code.

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: Changes requested from PR What’s changed: README.md bin composer.json composer.lock config docker-compose.yml docs env.default env.local manifest-egress.yml manifest.yml modules package-lock.json patches phpunit.xml.dist s3 scripts sp.crt sp.key storage themes usagov.sql vendor web Changed /es/indice-agencias-estatales, to /es/gobiernos-estatales for Agencias estatales README.md bin composer.json composer.lock config docker-compose.yml docs env.default env.local manifest-egress.yml manifest.yml modules package-lock.json patches phpunit.xml.dist s3 scripts sp.crt sp.key storage themes usagov.sql vendor web Changed regex for states and agencies from /agencies/* and /states/* to /agencies/.+ and /states/.+ README.md bin composer.json composer.lock config docker-compose.yml docs env.default env.local manifest-egress.yml manifest.yml modules package-lock.json patches phpunit.xml.dist s3 scripts sp.crt sp.key storage themes usagov.sql vendor web It was matching on an empty string before which is why the states was showing double and there was the clause for agencies README.md bin composer.json composer.lock config docker-compose.yml docs env.default env.local manifest-egress.yml manifest.yml modules package-lock.json patches phpunit.xml.dist s3 scripts sp.crt sp.key storage themes usagov.sql vendor web Combined lines 21 and 24

IsabelLaurenceau committed:
USAGOV-429-states-left-nav: Correction for breadcrumb link

IsabelLaurenceau commented:
What’s changed:
* Changed /es/indice-agencias-estatales, to /es/gobiernos-estatales for Agencias estatales
* Changed regex for states and agencies from /agencies/* and /states/*  to  /agencies/.+ and /states/.+ 
    * It was matching on an empty string before which is why the states was showing double and there was the clause for agencies 
* Combined lines 21 and 24 
* Corrected breadcrumb link for Agencias estatales

akf commented:
Something is wrong on my local dev, but I think it's because I didn't set up the Spanish test data correctly. I will finish reviewing tomorrow!

akf reviewed (approved):
I am still a bit confused by the Spanish menus -- I'm pretty sure there is an extra level showing up when I look at an /es/agencias/something page. 

But, we're going to need to update these menus anyway, as the content team is still building out this section and changing the order of things in the menu items. I think we should merge this in update the menus later. The right versions of the menus are appearing on the right pages, as far as I can tell.

akf committed:
Merge branch 'dev' into USAGOV-429-states-left-nav
